Gems/Atom/RPI/Code/Source/RPI.Public/Material/Material.cpp

```diff
@@ -820,5 +834,13 @@ namespace AZ
             return m_materialProperties.GetMaterialPropertiesLayout();
         }
 
+        // AssetBus overrides...
+        void Material::OnAssetReady(Data::Asset<Data::AssetData> asset)
+        {
+            Data::AssetBus::Handler::BusDisconnect();
```

Comment:
Q: Does this mean that Material intentionally shouldn't listen for and handle any OnAssetReload() notifications?

Reply:
That was surprising for me to find that, but I assume there are other systems at play that reload materials. @gadams3 may know something about it.

Reply:
For the most part, the rest of rendering system uses CanCompile(), and that protects against bad uses and retries later until CanCompile() returns true.

And yes, I agree with you that Atom should not care about Asset API issues. But I noticed, for example that there are other systems like EditorComponentModeFeedback just loads a material and assumes it is good to go. It will require a cleanup campaign across the board.